### PR TITLE
Add support and doc for AD579x DAC

### PR DESCRIPTION
**Pull request by SaikiranGudla**
Add precision toolbox support for AD5760, AD5780,
AD5781, AD5790, AD5791 DACs

### CODE REVIEW

File: +adi/+AD5760/Tx.m [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    %   tx = adi.AD5760.Tx;
+    %   tx = adi.AD5760.Tx('uri','ip:192.168.2.1');
+    %
+    %   <a href="https://www.analog.com/media/en/technical-documentation/data-sheets/ad5760.pdf">AD5760 Datasheet</a>
+
+    properties 
+        SampleRate
+        CodeSelect
+        PowerDown
+        Raw
```
Comment (tfcollins) [inline +adi/+AD5760/Tx.m:16]: All properties should have a default value with docstrings

File: +adi/+AD5760/Tx.m [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            obj.EnabledChannels = 1;
+            obj.uri = 'ip:analog.local';
+            obj.DataSource = 'DMA';
+        end
+
+        function set.SampleRate(obj, value)
+            % Set device sampling rate
+            obj.SampleRate = value;
+            if obj.ConnectedToDevice
+                obj.setDeviceAttributeRAW('sampling_frequency', num2str(value));            end
```
Comment (tfcollins) [inline +adi/+AD5760/Tx.m:60]: Can you move the "end" to the next line

File: +adi/+AD5760/Tx.m [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    %
+    %   tx = adi.AD5760.Tx;
+    %   tx = adi.AD5760.Tx('uri','ip:192.168.2.1');
+    %
+    %   <a href="https://www.analog.com/media/en/technical-documentation/data-sheets/ad5760.pdf">AD5760 Datasheet</a>
+
+    properties 
+        SampleRate
+        CodeSelect
+        PowerDown
```
Comment (tfcollins) [inline +adi/+AD5760/Tx.m:15]: This should be a Logical

File: +adi/+AD5760/Tx.m [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                id = 'voltage0';
+                obj.setDeviceAttributeRAW(id, 'code_select', num2str(value), true)
+            end
+        end
+
+        function set.PowerDown(obj, value)
+            options = [0,1];
+            if ~any(value==options)
+               error(['PowerDown must be one of ',num2str(options)]);
+            end
```
Comment (tfcollins) [inline +adi/+AD5760/Tx.m:76]: If you define PowerDown as a Logical this check is done for you.

**Review by tfcollins — CHANGES_REQUESTED** [review]
All the child classes are almost identical except for the device names. You should instead put all the properties in the base class an update the devName+phyDevName in the constructor calls. This greatly simplifies the code

File: +adi/+AD5781/Tx.m [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+        function set.SampleRate(obj, value)
+            % Set device sampling rate
+            obj.SampleRate = value;
+            if obj.ConnectedToDevice
+                obj.setDeviceAttributeRAW('sampling_frequency', num2str(value));
+            end
+        end
+
+                function set.CodeSelect(obj, value)
```
Comment (ribdp) [inline +adi/+AD5781/Tx.m:64]: Indentation is a bit off

**Comment by SaikiranGudla** [comment]
> All the child classes are almost identical except for the device names. You should instead put all the properties in the base class an update the devName+phyDevName in the constructor calls. This greatly simplifies the code

Yes, except for the device names, and resolution (data type), all the generics are almost identical. Do you suggest using a single Tx class for all the generics with the device name as an argument?

**Comment by SaikiranGudla** [comment]
v1.0: Addressed review comments

1. Moved all the properties to the base class and updated the devName, phyDevName, and dataType in the constructor calls
2. Defined PowerDown as logical
3. Added comments for properties

File: +adi/+AD579x/Base.m [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    % AD5791 is single channel 20bit DAC
+    % AD5780 is single channel 18bit DAC
+    % AD5781 is single channel 18bit DAC
+    % AD5760 is single channel 16bit DAC
+
+    properties(Nontunable)
+        % SamplesPerFrame Samples Per Frame
+        %   Number of samples per frame, specified as an even positive
+        %   integer.        
+         SamplesPerFrame
```
Comment (tfcollins) [inline +adi/+AD579x/Base.m:15]: Please provide a default value

**Review by tfcollins — CHANGES_REQUESTED** [review]
Just update the SamplesPerFrame and this is ready to merge

**Comment by SaikiranGudla** [comment]
v 1.1: Addressed review comments

Added default value for SamplesPerFrame

**Pull request merged**